### PR TITLE
Implement displaying of the card insertion

### PR DESCRIPTION
**Pull request by emaxx-google**
Ivan, PTAL.

This is still not the final version (a bunch of TODOs and also not sure about memory leaks), but there's already quite some stuff to be reviewed.

### CODE REVIEW

File: third_party/pcsc-lite/naclport/js_client/src/api.js [hunk 1]
```diff
@@ -2156,19 +2156,20 @@ API.ResultOrErrorCode.prototype.logger = GSC.Logging.getScopedLogger(
```
Comment (emaxx-google) [inline third_party/pcsc-lite/naclport/js_client/src/api.js]: The changes in this file are mostly mechanical.

File: smart_card_connector_app/src/window.css [hunk 2]
```diff
@@ -133,22 +133,26 @@ body {
...
   background: red;
 }
 
-.color-success {
+.status-indicator-success {
   background: green;
 }
+
+.status-indicator-success-with-reader {
+  background: navy;
```
Comment (emaxx-google) [inline smart_card_connector_app/src/window.css]: Haven't yet decided how to draw this state.

File: third_party/pcsc-lite/naclport/js_client/src/context.js [hunk 3]
```diff
@@ -115,23 +116,33 @@ goog.exportProperty(Context.prototype, 'logger', Context.prototype.logger);
 /**
  * Starts the initialization process by initiating a connection to the server
  * App.
+ *
+ * If the optional message channel is passed, then it is used instead of opening
```
Comment (emaxx-google) [inline third_party/pcsc-lite/naclport/js_client/src/context.js]: Turns out, it's not allowed to open a port from a background page to itself. So I had to write a fake message channel and extend this class to be able to receive it instead of opening the port.

File: common/js/src/messaging/message-channel-pair.js [hunk 4]
```diff
@@ -0,0 +1,83 @@
+/** @license
```
Comment (emaxx-google) [inline common/js/src/messaging/message-channel-pair.js]: See the change below in context.js for the reasoning.

File: third_party/pcsc-lite/naclport/server/src/reader-tracker.js [hunk 5]
```diff
@@ -150,81 +281,391 @@ ReaderTracker.prototype.readerFinishAddListener_ = function(message) {
...
-  var text =
-      'Finished initializing reader ' + name + ' (port ' + port +
-      ', device ' + device + ') with return code ' + returnCodeHex;
+  var logMessage =
+      'The reader "' + name + '" (port ' + port + ', device "' + device +
+      '") was initialized with return code ' + returnCodeHex;
+  if (returnCode === 0)
+    this.logger_.info(logMessage);
+  else
+    this.logger_.warning(logMessage);
```
Comment (ivansandrk) [inline third_party/pcsc-lite/naclport/server/src/reader-tracker.js]: Use a consistent bracing style (a few lines below you use braces, and here you don't).

File: common/js/src/logging/debug-dump.js [hunk 6]
```diff
@@ -183,7 +183,8 @@ function dump(value) {
   // Note that goog.dom.is* methods are not used because many of them may
   // produce thorny false positives.
   //
-  // TODO(emaxx): Think about a proper solution that deals with cyclic references.
+  // TODO(emaxx): Think about a proper solution that deals with cyclic
```
Comment (emaxx-google) [inline common/js/src/logging/debug-dump.js]: Sorry, committed these unrelated changes by mistake. But there's not much of them anyway.

File: common/js/src/popup-window/server.js [hunk 7]
```diff
@@ -65,7 +65,8 @@ GSC.PopupWindow.Server.createWindow = function(
 
   logger.fine(
       'Creating a popup window with url="' + url + '", options=' +
-      GSC.DebugDump.debugDump(createWindowOptions));
+      GSC.DebugDump.debugDump(createWindowOptions) + ', data=' +
```
Comment (emaxx-google) [inline common/js/src/popup-window/server.js]: Returned this logging back, as the recent changes made the debug dumping functions more robust.

File: smart_card_connector_app/src/background.js [hunk 8]
```diff
@@ -98,9 +104,13 @@ function launchedListener() {
 }
 
 function openWindow() {
-  var data = {'clientAppListUpdateSubscriber':
-      messageChannelPool.addOnUpdateListener.bind(messageChannelPool),
-      'readerTracker': readerTracker};
```
Comment (emaxx-google) [inline smart_card_connector_app/src/background.js]: I missed this when reviewing the original code, but passing class instances between windows is unsafe, given the presence of compiler optimizations.

File: smart_card_connector_app/src/window.css [hunk 9]
```diff
@@ -133,22 +133,26 @@ body {
...
+.status-indicator-failure {
   background: red;
 }
 
-.color-success {
+.status-indicator-success {
   background: green;
 }
+
+.status-indicator-success-with-reader {
```
Comment (ivansandrk) [inline smart_card_connector_app/src/window.css]: status-indicator-success-with-card ?

**Comment by ivansandrk** [comment]
lgtm

File: smart_card_connector_app/src/window.css [hunk 10]
```diff
@@ -133,22 +133,26 @@ body {
   align-items: center;
 }
 
-.error-indicator {
+.status-indicator {
   width: 9px;
   height: 9px;
   border-radius: 50%;
   display: flex;
```
Comment (emaxx-google) [inline smart_card_connector_app/src/window.css]: I'm going to remove these "flex" declarations, as, first, I don't understand how they work and how to adopt them for more complex cases, and, second, because they seem to adjust the font size too (while the font sizes were carefully chosen previously together with the UI team).

File: smart_card_connector_app/src/window.css (hunk 9 above)
Comment (emaxx-google) [inline smart_card_connector_app/src/window.css]: I finally named it "reader-card-present-indicator".
